alibi_detect/od/_gmm.py

```diff
@@ -0,0 +1,214 @@
+from typing import Union, Optional, Dict, Any, TYPE_CHECKING
+
+import numpy as np
+
+from alibi_detect.utils._types import Literal
+from alibi_detect.base import outlier_prediction_dict
+from alibi_detect.base import BaseDetector, ThresholdMixin, FitMixin
+from alibi_detect.od.pytorch import GMMTorch
+from alibi_detect.od.sklearn import GMMSklearn
+from alibi_detect.utils.frameworks import BackendValidator
+from alibi_detect.version import __version__
+from alibi_detect.exceptions import _catch_error as catch_error
+
+
+if TYPE_CHECKING:
+    import torch
+
+
+backends = {
+    'pytorch': GMMTorch,
+    'sklearn': GMMSklearn
+}
+
+
+class GMM(BaseDetector, ThresholdMixin, FitMixin):
+    def __init__(
+        self,
+        n_components: int = 1,
+        backend: Literal['pytorch', 'sklearn'] = 'pytorch',
```

Comment:
I reckon we probably go with sklearn backend as default. It's nice to have the pytorch variant for speedup on large datasets but it does ideally involve a little more thought around the fitting parameters which it seems silly to burden users with by default.